Adopt GHA auto-release workflow

Just like this:

https://github.com/metanorma/mn2pdf-ruby/blob/main/.github/workflows/release.yml

The variable names for secrets are identical.

Tried to bump new release, got error:
https://github.com/metanorma/nist-pubid/runs/4461445704?check_suite_focus=true
```
bump nist_pubid 0.1.3 0.1.3
version lib/nist_pubid/version.rb 0.1.3 0.1.3
git_add lib/nist_pubid/version.rb
git_commit "Bump nist_pubid to 0.1.3"
HEAD detached at 87d46c6
Untracked files:
  (use "git add <file>..." to include in what will be committed)
	.bundle/
	Gemfile.lock
	vendor/

nothing added to commit but untracked files present (use "git add" to track)
"The command `git commit -m "Bump nist_pubid to 0.1.3"` was unsuccessful. Aborting."
Error: Process completed with exit code 1.
```

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -16,7 +16,7 @@ jobs:
   release:
     runs-on: ubuntu-18.04
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
 
       - uses: ruby/setup-ruby@v1
         with:

